lib/google/ads/google_ads/logging_interceptor.rb

```diff
@@ -30,42 +30,54 @@ class LoggingInterceptor < GRPC::ClientInterceptor
         }
 
         def initialize(logger)
+          # Don't propogate args, parens are necessary
           super()
           @logger = logger
         end
 
         def request_response(request:, call:, method:, metadata: {})
           begin
             response = yield
-            response_message = sprintf("Incoming response: Payload: %s",
-                                       response.to_json)
 
             @logger.info(build_summary_message(request, call, method, false))
             @logger.debug(build_request_message(metadata, request))
-            @logger.debug(response_message)
-            return response
+            @logger.debug(build_success_response_message(response))
+            response
           rescue Exception
-            response_message = "Incoming response (errors): \n"
-
-            most_recent_error = Google::Gax::GaxError.new('')
-            most_recent_error.status_details && most_recent_error.status_details.each do |detail|
-              if INTERESTING_ERROR_CLASSES.include?(detail.class)
-                response_message = add_response_message_from_detail(
-                  response_message,
-                  detail,
-                )
-              end
-            end
-
             @logger.warn(build_summary_message(request, call, method, true))
             @logger.info(build_request_message(metadata, request))
-            @logger.info(response_message)
+            @logger.info(build_error_response_message)
             raise
           end
         end
 
         private
 
+        def build_error_response_message
+          # this looks like "magic", but the Google::Gax::GaxError grabs
+          # the current exception as its cause, and then parses details
+          # out of that exception. So this sets it up so that
+          # most_recent_error.status_details contains useful information about
+          # our failure
+          most_recent_error = Google::Gax::GaxError.new('')
+          response_message = "Incoming response (errors): \n"
+
+          return response_message unless most_recent_error.status_details
+
+          formatted_details = most_recent_error.status_details.select { |detail|
```

Comment:
We have another bug open because status_details can sometimes be a string. Can we address that as well while we're here?

Reply:
👍